### PR TITLE
`DifferenceToLower/Upper` forward declaration

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add `*.def` file for `DifferenceToLower` and `DifferenceToUpper`
- update code style (2nd commit)

### CODE REVIEW

File: include/picongpu/algorithms/DifferenceToLower.def [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+#include <pmacc/math/Vector.hpp>
+
+
+namespace picongpu
+{
+
+    /** calculate difference to lower value
+     *
+     * @tparam T_Dim for how many dimensions this operator access memory
```
Comment (PrometheusPi) [inline include/picongpu/algorithms/DifferenceToLower.def]: REQUIRED: 
The wording is unclear 
do you mean
number of dimensions the operator accesses in memory
?

File: include/picongpu/algorithms/DifferenceToLower.def [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     *
+     * If `GetDifference` is called for a direction greater equal T_Dim always
```
Comment (PrometheusPi) [inline include/picongpu/algorithms/DifferenceToLower.def]: OPTIONAL:
greater **or** equal **to**

File: include/picongpu/algorithms/DifferenceToLower.def [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+     * a zeroed value is returned
```
Comment (PrometheusPi) [inline include/picongpu/algorithms/DifferenceToLower.def]: QUESTION:
what do you mean by that?

File: include/picongpu/algorithms/DifferenceToUpper.def [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace picongpu
+{
+
+    /** calculate difference to upper value
+     *
+     * @tparam T_Dim for how many dimensions this operator access memory
+     *
+     * If `GetDifference` is called for a direction greater equal T_Dim always
+     * a zeroed value is returned
```
Comment (PrometheusPi) [inline include/picongpu/algorithms/DifferenceToUpper.def]: same as above

File: include/picongpu/algorithms/DifferenceToLower.hpp [hunk 5]
```diff
@@ -28,71 +28,86 @@
...
+            {
+            }
 
-        HDINLINE GetDifference()
-        {
-        }
+            /** @return always a zeroed value
+             */
+            template< typename Memory >
+            HDINLINE typename Memory::ValueType operator()( Memory const & mem) const
```
Comment (PrometheusPi) [inline include/picongpu/algorithms/DifferenceToLower.hpp]: QUESTION:
is there a space missing after `mem` ?

File: include/picongpu/algorithms/DifferenceToLower.def [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    template< uint32_t T_Dim >
+    struct DifferenceToLower;
+
+} //namespace picongpu
```
Comment (ax3l) [inline include/picongpu/algorithms/DifferenceToLower.def]: space :)

**Pull request merged**